contrib/utilities/check_encoding.py

```diff
@@ -21,16 +21,23 @@
 # repository:
 # contrib/utilities/check_encoding.py
 
-import glob
 import itertools
+import os
 
-filenames = itertools.chain(glob.iglob('**/*.h', recursive=True),
-                            glob.iglob('**/*.cc', recursive=True),
-                            glob.iglob('**/*.html', recursive=True))
+def filename_generator(suffix):
+    for root, _, file_names in os.walk("./"):
+        for file_name in file_names:
+            if file_name.endswith(suffix):
+                yield "./" + root + "/" + file_name
+
+
+filenames = itertools.chain(filename_generator(".h"),
+                            filename_generator(".cc"),
+                            filename_generator(".html"))
 
 for filename in filenames:
     try:
-        with open(filename, encoding='utf-8') as file:
-            file.read()
+        with open(filename, encoding='utf-8') as file_handle:
```

Comment:
If you want to use the parameter `encoding` in python2 then you have to use `io.open()`. `io.open()` works the same way for python2 and python3.

If you use `open()` with the parameter `encoding` an error is supposed to be raised.

https://docs.python.org/2/library/io.html
https://docs.python.org/2/library/functions.html#open

Reply:
You're right. I thought I tested this with python2 and python3 but apparently not.